2.0/x-oca-event.md

```diff
@@ -2,47 +2,55 @@
+| host_ref                 | `object-ref`                | references the host related to this event. must be of type `x-oca-asset`                            |
+| url_ref                  | `object-ref`                | references a url related to this event. must be of type `url`                                       |
+| file_ref                 | `object-ref`                | references a file related to this event. must be of type `file`                                     |
+| process_ref              | `object-ref`                | references a process related to this event. must be of type `process`                               |
+| parent_process_ref       | `object-ref`                | references a parent process related to this event. must be of type `process`                        |
+| cross_process_target_ref | `object-ref`                | references the target process in case of a cross process event. must be of type `process`           |
+| domain_ref               | `object-ref`                | references a domain related to this event. must be of type `domain-name`                            |
+| registry_ref             | `object-ref`                | references windows registry data related to this event. must be of type `windows-registry-key`      |
+| network_ref              | `object-ref`                | references network traffic related to this event. must be of type `network-traffic`                 |
+| ip_refs                  | `list` of type `object-ref` | references any ip addresses related to this event. must be of type `ipv4-address` or `ipv6-address` |
+| user_ref                 | `object-ref`                | references the user related to this event. must be of type `user-account`                           |
+| severity                 | `number`                    | Severity of the event on a scale of 1-100.                                                          | 
+| start                    | `timestamp`                 | The start time in UTC of the event.                                                                 |
+| end                      | `timestamp`                 | The end time in UTC of the event.                                                                   |
+| timezone                 | `string`                    | The timezone where the event was discovered.                                                        |
+| duration                 | `string`                    | The duration of the event.                                                                          |
+| dataset                  | `string`                    | The dataset of the event.                                                                           |
+
+### Event Specific properties
+Certain events are important for security context which have their unique properties. The `x-oca-event` supports the following optional properties that should be populated only in certain `action` types.
```

Comment:
If the below properties should only be added with certain action types, we may need to better describe what these `action` types are.

Reply:
I think the description is fine. Do you have an example of a security product that would return data that could be mapped to this property?

Reply:
the exact name of the action is impossible to depict since it differs between sources. one EDR will call it `NamedPipeEvent` while the other `PipeEvent`.
It is up to whoever creates the event to understand the context and populate the field or not.
At least until we have event action normalized for all stix producers